app/controllers/api/audit_logs_controller.rb

```diff
@@ -0,0 +1,27 @@
+class Api::AuditLogsController < Api::BaseController
+  before_action :authorize_audit_log, only: :index
+
+  def index
+    if current_user.admin?
+      audit_logs = AuditLog.all
+      audit_logs = audit_logs.where('timestamp >= ? AND timestamp <= ?', date_range[:start_date], date_range[:end_date]) if params[:dateRange].present?
+      audit_logs = audit_logs.where(user_id: params[:userId]) if params[:userId].present?
```

Comment:
The `date_range` method does not handle invalid date formats or ranges. Consider adding validation to ensure the dates are in the correct format and that the start date is not after the end date.